### ISSUE TITLE
Path and mypy

### ISSUE THREAD

**Issue by socketpair** [python/mypy#10238]
This example must not trigger errors in MyPy

```python
from pathlib import PosixPath

class AtomicPath(PosixPath):
    @staticmethod
    def test():
        print('test')


x = list(AtomicPath('/etc/test').parent.glob('*'))

print(repr(x[0]))  # prints AtomicPath('/etc/mtab')  (!!!!)

# MyPy considers x[0] is Path, but actually AtomicPath  (!!!)
x[0].test()  # bug.py:14: error: "Path" has no attribute "test"
```

**Comment by ethanhs** [comment]
What mypy version are you on? I cannot repro this on 0.812 or 0.800. I believe you should update.

**Comment by socketpair** [comment]
@ethanhs 
```
$ rpm -qa | egrep 'mypy|shed'
python3-mypy_extensions-0.4.3-4.fc33.noarch
python-typeshed-0.1-0.20191011git2.fc33.noarch
python3-mypy-0.782-2.fc33.noarch
```

**Comment by ethanhs** [comment]
Okay, looks like this is no longer an issue then.

**Comment by socketpair** [comment]
@ethanhs did you test or just guess ?
